clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -213,8 +214,12 @@ public synchronized void addRecord(ConsumerRecord<K, V> record) {
         recs.add(record);
     }
 
-    public synchronized void setException(KafkaException exception) {
-        this.exception = exception;
```

Comment:
@pgwhalen, after thinking about this some more, I think we should keep this older signature, deprecate it, and have it call `setPollException(exception)`. Although `MockConsumer` is strictly speaking a non-public API, doing this will help avoid problems should people use the `MockConsumer` class in their tests. And it'd probably help to add JavaDoc that says to use `setPollException(...)` instead.

Reply:
Good point, I forgot it was a mildly public API.  I added it back, should be good to merge now unless anything else stands out.